packages/aws-cdk/bin/cdk.ts

```diff
@@ -347,14 +348,78 @@ async function initCommandLine() {
 
     fs.mkdirpSync(outputDir);
 
+    const extension = json ? 'json' : 'yaml';
+    const hashCache: { [path: string]: string } = {};
     for (const stack of stacks) {
       const finalName = renames.finalName(stack.name);
-      const fileName = `${outputDir}/${finalName}.template.${json ? 'json' : 'yaml'}`;
+      const fileName = path.join(outputDir, `${finalName}.template.${extension}`);
       highlight(fileName);
       await fs.writeFile(fileName, toJsonOrYaml(stack.template));
+      const manifestName = path.join(outputDir, `${finalName}.manifest.${extension}`);
+      await fs.writeFile(manifestName, toJsonOrYaml(await prepareManifest(stack, outputDir, hashCache)));
     }
 
     return undefined; // Nothing to print
+
+    /**
+     * Prepares an asset manifest document, staging assets in a specified directory.
+     * @param stack the stack for which assets are to be manifested.
+     * @param stackName the name of the stack (accounting for renames)
+     * @param dir the directory under which to stage the assets (an assets sub-directory will be used).
+     * @param cache a cache for asset files hashes, improving performance when the same assets are used across multiple stacks
+     */
+    async function prepareManifest(stack: cxapi.SynthesizedStack, dir: string, cache: { [path: string]: string }) {
```

Comment:
Interesting... My intuition was that all the assets and asset manifest should be emitted to the working directory by the `App` and not by the toolkit. Now that the cx protocol enables this, I wonder if it doesn't make more sense to move all this logic to `App` so that we'll have less magic in the toolkit.

What do you think?

I am okay if you feel this can be done in two steps, but it would be nice to have some agreement on where we want this to go, so we'll not take too many redundant steps.

Reply:
That actually sounds like it'd be better. I had started this before you made the changes to the `App` API, and I missed the fact I could refactor this away to the other side of the wall.